tasks/main.yml

```diff
@@ -66,6 +66,18 @@
         value: replaced
   register: firewall_lib_result
 
+- name: gather firewall ansible fact
+  block:
+    - name: gather firewall config information
+      firewall_lib_facts:
+      register: __firewalld_facts
+
+    - name: update firewalld_config fact
+      set_fact:
+        firewall_config: >-
+          "{{ __firewalld_facts.ansible_facts.firewalld_config }}"
+  when: firewall == None
```

Comment:
I think you should use the boolean here - `when: not firewall | bool` - just in case someone tries `firewall: {}` or `firewall: []`

Reply:
Yeah that is what I started with, but `firewall = [{previous: replaced}]` makes it through the conditional `not firewall | bool` for no good reason.

Reply:
Note: I don't think it is only `previous` that causes this issue, but that is what I use to test the conditional at the moment

Reply:
ok - yes - I guess in this particular context, a non-empty dict evaluates to boolean `False` for some reason
then use
```yaml
  when: firewall is none or firewall | length == 0
```
note that `when` (and other conditionals such as `that`, `failed_when`, `changed_when`, etc.) are evaluated as Jinja expressions (that is, you don't have to use `{{ ... }}`) - use lowercase `none` - https://jinja.palletsprojects.com/en/2.9.x/templates/#literals (not sure if this is true for jinja 2.7, but I believe it works)